pkg/recipes/terraform/config/providers/azure.go

```diff
@@ -119,30 +120,62 @@ func fetchAzureCredentials(ctx context.Context, azureCredentialsProvider credent
 	credentials, err := azureCredentialsProvider.Fetch(ctx, credentials.AzureCloud, "default")
 	if err != nil {
 		if errors.Is(err, &secret.ErrNotFound{}) {
-			logger.Info("AWS credentials are not registered, skipping credentials configuration.")
+			logger.Info("Azure credentials are not registered, skipping credentials configuration.")
 			return nil, nil
 		}
 
 		return nil, err
 	}
 
-	if credentials == nil || credentials.ClientID == "" || credentials.TenantID == "" || credentials.ClientSecret == "" {
+	switch credentials.Kind {
+	case ucp_datamodel.AzureServicePrincipalCredentialKind:
+		if credentials.ServicePrincipal == nil ||
+			credentials.ServicePrincipal.ClientID == "" ||
+			credentials.ServicePrincipal.TenantID == "" ||
+			credentials.ServicePrincipal.ClientSecret == "" {
+			logger.Info("Azure service principal credentials are not registered, skipping credentials configuration.")
+			return nil, nil
+		}
+
+		return credentials, nil
+	case ucp_datamodel.AzureWorkloadIdentityCredentialKind:
+		if credentials.WorkloadIdentity == nil ||
+			credentials.WorkloadIdentity.ClientID == "" ||
+			credentials.WorkloadIdentity.TenantID == "" {
+			logger.Info("Azure workload identity credentials are not registered, skipping credentials configuration.")
+			return nil, nil
+		}
+
+		return credentials, nil
+	default:
 		logger.Info("Azure credentials are not registered, skipping credentials configuration.")
 		return nil, nil
 	}
 
-	return credentials, nil
 }
 
 func (p *azureProvider) generateProviderConfigMap(configMap map[string]any, credentials *credentials.AzureCredential, subscriptionID string) map[string]any {
 	if subscriptionID != "" {
 		configMap[azureSubIDParam] = subscriptionID
 	}
 
-	if credentials != nil && credentials.ClientID != "" && credentials.TenantID != "" && credentials.ClientSecret != "" {
-		configMap[azureClientIDParam] = credentials.ClientID
-		configMap[azureClientSecretParam] = credentials.ClientSecret
-		configMap[azureTenantIDParam] = credentials.TenantID
+	switch credentials.Kind {
+	case ucp_datamodel.AzureServicePrincipalCredentialKind:
+		if credentials.ServicePrincipal != nil &&
+			credentials.ServicePrincipal.ClientID != "" &&
+			credentials.ServicePrincipal.TenantID != "" &&
+			credentials.ServicePrincipal.ClientSecret != "" {
+			configMap[azureClientIDParam] = credentials.ServicePrincipal.ClientID
+			configMap[azureClientSecretParam] = credentials.ServicePrincipal.ClientSecret
+			configMap[azureTenantIDParam] = credentials.ServicePrincipal.TenantID
+		}
+	case ucp_datamodel.AzureWorkloadIdentityCredentialKind:
+		if credentials.WorkloadIdentity != nil &&
+			credentials.WorkloadIdentity.ClientID != "" &&
+			credentials.WorkloadIdentity.TenantID != "" {
+			configMap[azureClientIDParam] = credentials.WorkloadIdentity.ClientID
+			configMap[azureTenantIDParam] = credentials.WorkloadIdentity.TenantID
```

Comment:
So looks like for workload identity a few additional properties are expected by terraform - https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/aks_workload_identity#configuring-terraform-to-use-an-aks-workload-identity

```
use_aks_workload_identity = true
use_cli                   = false
```